DataBiosphere/consent-ui
# GAWB-2907: Fix build errors from old versions

No Jira, but required before https://github.com/broadinstitute/consent-ui/pull/161 is merged, so this is a pre-req for GAWB-2578

Current build uses a very old version of node and fails with recent upgrades to the npm request module. 

# Changes: 
* Upgrade to latest stable node
* Fix bower metadata complaint about the name
* Update karma to the latest fixed version
* Fixed a closing anchor tag that manifested as strange pagination behavior

# Testing:
Two things need to happen for testing, and a third for functional testing.

1. Build the docker image. Inside consent-ui, run `docker build .` and the image should build with no errors. Lots of warnings though.
2. Run the build script that allows you to run your changes locally. Inside consent-ui, run `sudo ./build.sh` and again, you should not see any errors. Lots of warnings though.
3. Lastly, to run the built image, generate configs as described in the readme. Optionally, change the build image to be one that you build locally with the docker build command above (see bullet 1, but build with a tag so you can use it). Once running, you can go to https://local.broadinstitute.org/ and click around. No JS errors should happen.
